schema/auction_result.js

```diff
@@ -0,0 +1,80 @@
+import date from "./fields/date"
+import { amount } from "./fields/money"
+import { IDFields, NodeInterface } from "./object_identification"
+import { GraphQLFloat, GraphQLNonNull, GraphQLString, GraphQLObjectType } from "graphql"
+import { connectionDefinitions } from "graphql-relay"
+import { has } from "lodash"
+
+const AuctionResultType = new GraphQLObjectType({
+  name: "AuctionResult",
+  interfaces: [NodeInterface],
+  isTypeOf: obj => has(obj, "dimension_text") && has(obj, "organization"),
+  fields: () => ({
+    ...IDFields,
+    title: {
+      type: GraphQLString,
+    },
+    artist_id: {
+      type: new GraphQLNonNull(GraphQLString),
+    },
+    date,
+    date_text: {
+      type: GraphQLString,
+    },
+    medium_text: {
+      type: GraphQLString,
+    },
+    category_text: {
+      type: GraphQLString,
+    },
+    dimension_text: {
+      type: GraphQLString,
+    },
+    dimensions: {
+      type: new GraphQLObjectType({
+        name: "AuctionLotDimensions",
+        description: "In centimeters.",
+        fields: {
+          width: {
+            type: GraphQLFloat,
+          },
+          height: {
+            type: GraphQLFloat,
+          },
+          depth: {
+            type: GraphQLFloat,
+          },
+        },
+      }),
+      resolve: ({ width_cm, height_cm, depth_cm }) => {
+        return { width: width_cm, height: height_cm, depth: depth_cm }
+      },
+    },
+    organization: {
+      type: GraphQLString,
+    },
+    sale_date: date,
+    sale_date_text: {
+      type: GraphQLString,
+    },
+    sale_title: {
+      type: GraphQLString,
+    },
+    currency: {
+      type: GraphQLString,
+    },
+    description: {
+      type: GraphQLString,
+    },
+    external_url: {
+      type: GraphQLString,
+    },
+    low_estimate: amount(({ low_estimate_cents_usd }) => low_estimate_cents_usd),
+    high_estimate: amount(({ high_estimate_cents_usd }) => high_estimate_cents_usd),
+    price_realized: amount(({ price_realized_cents_usd }) => price_realized_cents_usd),
```

Comment:
Will we always display in USD? If not you'll need the non-dollar denomination prices too (eg `low_estimate_cents`). Currently these prices can't be displayed with the above currency denominator.

Reply:
The existing Force (list/individual) views of auction results show native currencies so we should either model it here or follow up shortly. We should definitely _not_ name the fields ambiguously since we know we'll need both.

Reply:
Ah, I was just looking at the market insights comp, and not the current auction result pages.

In that case let's come back to this, the first portion of the market insights UI/schema we'll do is the 'Gallery Representation' one. When we get to the Auction Result one, we can just circle back to this. Nothing's using it in the meantime.

Reply:
Also, maybe all price related fields should go into a `price` or `pricing` object?

Reply:
I agree with all of these, but it's worth noting that whenever we return other 'price' related fields (like artwork pricing, or bid related amounts, etc.), we have a 'standard' schema we settled on: namely, we return `display_` fields that are already formatted, we always include `symbol` (so things can be custom formatted on the client), and we include `_cents` (in USD).

Here, we return `_cents` (in local), `_cents_usd` (in USD), and `currency` (3-letter code, _not_ the symbol).

While we can work with this...we have built in field types and helpers that generally work with that expected schema when dealing with 'money'. Similarly, for images we've settled on a standard schema (including `original_height`, `original_width`, etc.). Whereas the Diffusion API returns something slightly different for the image as well.

In all these cases, we can certainly work around that here, but it would be nice to be able to reuse schema fields/helpers (as we've generally been doing).

I think we should maybe pair up on finishing this PR (and getting the basic info in Reaction- we don't need to pair on more nitty gritty UI stuff)- that might illuminate why we should make some tweaks to the upstream API to better match other services and responses, or it might illuminate why something in Metaphysics is too brittle or not generic enough, and we can improve it here.

Either way...I'm really wanting to do a 'squad' approach to the Market Insights project, where we pair and regularly work on it (cc @joeyAghion ). I think that it would yield a better overall result, rather than someone independently adding an API, then a month later someone else independently trying to integrate it into the front-end, etc.

Thoughts?

Reply:
At the very least, it feels like Diffusion should possibly return the required `display_` fields?

Reply:
This is another pattern that was unclear at the API stage (and another potential benefit of my hope to build out from MP stubs). It would be great if these expectations for backing APIs were documented somewhere.

Separately, I actually think it might be an improvement if Metaphysics handled some of the "straightforward" UI-focused transformations, like converting currency amounts to their various displayable forms. Otherwise, the [sometimes complex] handling of these would need to be sprinkled across many different systems.

Reply:
> This is another pattern that was unclear at the API stage (and another potential benefit of my hope to build out from MP stubs). It would be great if these expectations for backing APIs were documented somewhere.

That's why we should squad!!

> Separately, I actually think it might be an improvement if Metaphysics handled some of the "straightforward" UI-focused transformations, like converting currency amounts to their various displayable forms. Otherwise, the [sometimes complex] handling of these would need to be sprinkled across many different systems.

I agree, but can go back and forth. It's probably correct that Metaphysics should be able to do its own formatting, which it actually can, in a rudimentary way. It uses http://openexchangerates.github.io/accounting.js/. One thing currently not possible, is exchanging a currency designation (`EUR` for instance), for the Euro symbol. Is this something the upstream provider should include, or should we add a more fully-featured 'money' lib to Metaphysics?